forms-flow-web/src/components/Modeler/SubflowCreateEdit.js

```diff
@@ -0,0 +1,456 @@
+    setIsPublished(processData.status === "Published");
+  }, [processData]);
+
+  const publishText = isPublished ? "Unpublish" : "Publish";
+
+  // get process name to dispaly
+  const processName = useMemo(() => {
+    if (!processData.processData) return;
+    return extractDataFromDiagram(processData?.processData).name;
+  }, [processData]);
+
+  //fetch process details using processkey
+  useEffect(async () => {
+    if (processKey) {
+      try {
+        setIsProcessDetailsLoading(true);
+        const { data } = await getProcessDetails(processKey);
+        setIsPublished(!isPublished);
+        dispatch(setProcessData(data));
+      } catch (error) {
+        console.error(error);
+      } finally {
+        setIsProcessDetailsLoading(false);
+      }
+    }
+  }, [processKey]);
+
+  //reset the data
+  useEffect(() => {
+    if (isCreate) {
+      dispatch(setProcessData({}));
+    }
+    return () => {
+      //if we click duplicate the the data will exist on the redux, so need to reset
+      isCreate &&
+        dispatch(setProcessDiagramXML(createNewProcess().defaultWorkflow.xml));
+    };
+  }, [isCreate]);
+
+  const handleToggleConfirmModal = () => setShowConfirmModal(!showConfirmModal);
+  const openConfirmModal = (type) => {
+    setModalType(type);
+    handleToggleConfirmModal();
+  };
+  const saveFlow = async (isPublishing = false) => {
+    try {
+      const bpmnModeler = bpmnRef.current?.getBpmnModeler();
+      const xml = await createXMLFromModeler(bpmnModeler);
+
+      if (!validateProcess(xml, lintErrors, t)) {
+        return;
+      }
+      if (!isCreate) {
+        const isEqual = await compareXML(processData?.processData, xml);
+        if (isEqual) {
+          !isPublishing && toast.success(t("Process is already up to date"));
+          return;
+        }
+      }
+
+      setSavingFlow(true);
+      const response = isCreate
+        ? await createProcess({ type: "BPMN", data: xml })
+        : await updateProcess({ type: "BPMN", id: processData.id, data: xml });
+
+      dispatch(setProcessData(response.data));
+      !isPublishing &&
+        toast.success(
+          t(`Subflow ${isCreate ? "created" : "updated"} successfully`)
+        );
+      if (isCreate && !isPublishing) {
+        dispatch(
+          push(`${redirectUrl}subflow/edit/${response.data.processKey}`)
+        );
+      }
+      return response.data;
+    } catch (error) {
+      toast.error(t("Failed to save process"));
+    } finally {
+      setSavingFlow(false);
+    }
+  };
+
+  const confirmPublishOrUnPublish = async () => {
+    try {
+      const bpmnModeler = bpmnRef.current?.getBpmnModeler();
+      const xml = await createXMLFromModeler(bpmnModeler);
+
+      if (!isPublished && !validateProcess(xml, lintErrors)) {
+        return;
+      }
+
+      const actionFunction = isPublished ? unPublish : publish;
+
+      const response = !isPublished ? await saveFlow(!isPublished) : null;
+
+      closeModal();
```

Comment:
will it be good to call closeModal after API call response is received? @abilpraju-aot

Reply:
here we handling two scenarios edit and create thats why called after response